igor-web/src/main/java/com/netflix/spinnaker/igor/concourse/service/ConcourseService.java

```diff
@@ -0,0 +1,246 @@
+    build.setFullDisplayName(job.getTeamName() + "/" + job.getPipelineName() + "/" + job.getName());
+    build.setUrl(host.getUrl() + "/teams/" + job.getTeamName() + "/pipelines/" + job.getPipelineName() + "/jobs/" +
+      job.getName() + "/builds/" + b.getNumber());
+
+    if (!fetchResources) {
+      return build;
+    }
+
+    Collection<Resource> resources = getResources(b.getId());
+
+    // merge input and output metadata into one map for each resource
+    Map<String, Map<String, String>> mergedMetadataByResourceName = resources.stream()
+      .collect(
+        groupingBy(Resource::getName,
+          reducing(emptyMap(), Resource::getMetadata,
+            (m1, m2) -> {
+              Map<String, String> m1OrEmpty = m1 == null ? emptyMap() : m1;
+              Map<String, String> m2OrEmpty = m2 == null ? emptyMap() : m2;
+              return Stream.concat(m1OrEmpty.entrySet().stream(), m2OrEmpty.entrySet().stream())
+                .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
+            }))
+      );
+
+    // extract git information from this particular named resource type
+    resources.stream().filter(r -> r.getType().equals("git"))
+      .map(Resource::getName)
+      .findAny()
+      .ifPresent(gitResourceName -> {
+        Map<String, String> git = mergedMetadataByResourceName.remove(gitResourceName);
+        if (git != null) {
+          String message = git.get("message");
+          String timestamp = git.get("committer_date");
+          build.setGenericGitRevisions(Collections.singletonList(GenericGitRevision.builder()
+            .committer(git.get("committer"))
+            .branch(git.get("branch"))
+            .message(message == null ? null : message.trim())
+            .sha1(git.get("commit"))
+            .timestamp(timestamp == null ? null : ZonedDateTime.parse(timestamp,
+              DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss Z")).toInstant())
+            .build()));
+        }
+      });
+
+    if (!mergedMetadataByResourceName.isEmpty()) {
+      build.setProperties(mergedMetadataByResourceName);
+    }
+
+    return build;
+  }
+
+  private Collection<Resource> getResources(String buildId) {
+    Map<String, Resource> resources = client.getBuildService().plan(buildId)
+      .getResources()
+      .stream()
+      .filter(r -> resourceFilter == null
+        || "git".equals(r.getType()) // there is a place for Git revision history on GenericBuild
+        || resourceFilter.matcher(r.getType()).matches())
+      .collect(toMap(Resource::getId, Function.identity()));
+
+    if (!resources.isEmpty()) {
+      setResourceMetadata(buildId, resources);
+    }
+
+    return resources.values();
+  }
+
+  /**
+   * Uses Concourse's build event stream to locate and populate resource metadata
+   */
+  private void setResourceMetadata(String buildId, Map<String, Resource> resources) {
+    Flux<Event> events = client.getEventService().resourceEvents(buildId);
```

Comment:
Seems like some tests for ConcourseService would be nice in the future. There is a lot going on here, specifically around handling edge cases.